pkg/vm/stackitem/item.go

```diff
@@ -538,25 +534,21 @@ func (i Bool) Convert(typ Type) (Item, error) {
 }
 
 // ByteArray represents a byte array on the stack.
-type ByteArray struct {
-	value []byte
-}
+type ByteArray []byte
 
 // NewByteArray returns an new ByteArray object.
 func NewByteArray(b []byte) *ByteArray {
-	return &ByteArray{
-		value: b,
-	}
+	return (*ByteArray)(&b)
 }
 
 // Value implements Item interface.
 func (i *ByteArray) Value() interface{} {
```

Comment:
Can pointer be avoided here?

Reply:
It certainly can, though I am not sure this will speed-up things: while there is 1 less indirection step, we will copy 2.5 as much data for every method.

Reply:
OK, let it be the way it is for now and we can always get back to it later.

Reply:
I'd expect it to be somewhat faster, but it just needs some testing.